Add support for Redis SINTERCARD command

Kiminni opened the pull request:
## Summary

Implements support for Redis 7.0+ SINTERCARD command in Spring Data Redis.

Fixes #2906

## Changes

### Core Implementation
- Add `sInterCard` method to `RedisSetCommands` interface
- Implement `sInterCard` in `JedisSetCommands` using Jedis client
- Implement `sInterCard` in `LettuceSetCommands` using Lettuce client

### Cluster Support  
- Add cluster-aware `sInterCard` in `JedisClusterSetCommands`
- Add cluster-aware `sInterCard` in `LettuceClusterSetCommands`
- Same-slot optimization with native cluster commands
- Cross-slot fallback using intersection calculation

### Connection Wrappers
- Add `sInterCard` delegation in `DefaultedRedisConnection`
- Add `sInterCard` support in `StringRedisConnection`
- Implement serialization handling in `DefaultStringRedisConnection`

### High-Level API
- Add user-friendly `intersectSize` methods to `SetOperations`
- Implement `intersectSize` in `DefaultSetOperations`
- Support multiple method overloads for different key patterns

### Testing
- Comprehensive integration tests with `@EnabledOnCommand("SINTERCARD")`
- Test coverage for all connection types (Jedis, Lettuce, Cluster)
- Various intersection scenarios (empty, partial, full)

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

onobc commented on src/main/java/org/springframework/data/redis/connection/RedisSetCommands.java:
```diff
@@ -153,6 +154,16 @@ public interface RedisSetCommands {
+	 * @param keys must not be {@literal null}.
+	 * @return {@literal null} when used in pipeline / transaction.
+	 * @see <a href="https://redis.io/commands/sintercard">Redis Documentation: SINTERCARD</a>
+	 * @since 3.4
```
I believe the `@since` should be 4.0 now (quite a bit of time has passed since initially submitted).

onobc reviewed (changes requested):
Thanks for the excellent contribution @Kiminni.

I believe there are still a few places to add the new API to:

- `RedisSet` / `DefaultRedisSet` / `DefaultRedisZSet`
- `BoundSetOperations`
- the Reactive counterparts

Kiminni commented:
Hello, @onobc.

I’ve applied the requested changes and addressed the review feedback.
Please take another look. If this approach looks good, feel free to merge; otherwise let me know what else you’d like adjusted.

Thank you!

onobc commented:
Thank you for the contribution @Kiminni - well done! I am closing in favor of https://github.com/spring-projects/spring-data-redis/commit/b09f75261c2042f2f0ab5438d0b56820d9b7e1ed
